orangecontrib/spectroscopy/widgets/gui.py

```diff
@@ -474,3 +477,152 @@ def set_default(self, v):
+        right = self.span[1]
+        #sets our line span to go from our minimum value to maximum
+        br.setLeft(left)
+        br.setRight(right)
+        br = br.normalized()
+
+        vs = self.getViewBox().size()
+        if self.mouseDragEvent:
+            left = self.span[0] - self.getYPos()
+
+        if self._bounds != br or self._lastViewSize != vs:
+            self._bounds = br
+            self._lastViewSize = vs
+            self.prepareGeometryChange()
+
+        self._endPoints = (left, right)
+        self._lastViewRect = vr
+
+        return self._bounds
+
+    def boundingRect(self):
+        if self._boundingRect is None:
+            self._boundingRect = self._computeBoundingRect()
+        return self._boundingRect
+
+    def paint(self, p, *args):
+        p.setRenderHint(p.Antialiasing)
+        left, right = self._endPoints
+        pen = self.currentPen
+        pen.setJoinStyle(QtCore.Qt.MiterJoin)
+        p.setPen(pen)
+        p.drawLine(Point(left, 0), Point(right, 0))
+
+
+    def mouseDragEvent(self, ev):
+        if self.movable and ev.button() == QtCore.Qt.LeftButton:
+            if ev.isStart():
+                self.moving = True
+                self.cursorOffset = self.pos() - self.mapToParent(ev.buttonDownPos())
+                self.startPosition = self.pos()
+            ev.accept()
+
+            if not self.moving:
+                return
+
+            self.setPos(self.cursorOffset + self.mapToParent(ev.pos()))
+            self.sigDragged.emit(self)
+            self.updateLabel()
+            if ev.isFinish():
+                self.moving = False
+                self.sigPositionChangeFinished.emit(self)
+                self._computeBoundingRect()
+
+    def mouseClickEvent(self, ev):
+        if self.moving and ev.button() == QtCore.Qt.RightButton:
+            ev.accept()
+            self.setPos(self.startPosition)
+            self.moving = False
+            self.sigDragged.emit(self)
+            self.sigPositionChangeFinished.emit(self)
+
+    def mouseDoubleClickEvent(self, ev):
+        if ev.button() == QtCore.Qt.RightButton:
+            ev.accept()
+            self.hide()
+            self.update()
+
+    def updateLabel(self):
+        x = self.getXPos()
+        leng = len(str(round(x)))
+        if leng >= 4:
+            leng = 2
+        elif leng < 4 and leng > 2:
+            leng = 3
+        elif leng <= 2:
+            leng = 4
+        self.label.setText(str(round(self.getXPos(), leng)))
+        self.update()
+
+class VerticalPeakLineLabel(pg.InfLineLabel):
+    def __init__(self, line, text="", movable=False, position=0.5, anchors=None, **kwds):
+        super().__init__(line)
+        self.line = line
+
+    def valueChanged(self):
+        if not self.isVisible():
+            return
+        value = self.line.value()
+        self.setText(self.format.format(value=value))
+        self.updatePosition()
+
+    def mouseDragEvent(self, ev):
```

Comment:
With a quick check I do not see any difference. What is different? If nothing, this function can be removed.